### ISSUE TITLE
BUG: Attempting to call gdf.estimate_utm_crs() on an empty GeoDataFrame kills Python (GeoPandas 0.9)

### ISSUE THREAD

**Issue by christaina** [geopandas/geopandas#1868]
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of geopandas.

---

#### Code Sample, a copy-pastable example

```python
import geopandas as gpd

gdf = gpd.GeoDataFrame([], columns={'geometry'}, crs='epsg:4326')
gdf.estimate_utm_crs()
```

#### Problem description

The python kernel dying makes it difficult to trace what causes this error. I think the expected behavior should be to raise an error.

#### Expected Output

I would expect this issue to raise an error rather than killing the kernel.

#### Output of ``geopandas.show_versions()``

<details>

SYSTEM INFO
-----------
python     : 3.6.12 | packaged by conda-forge | (default, Dec  9 2020, 00:24:39)  [GCC Clang 11.0.0]
executable : /Users/christina.bogdan/miniconda3/envs/ovl/bin/python
machine    : Darwin-19.6.0-x86_64-i386-64bit

GEOS, GDAL, PROJ INFO
---------------------
GEOS       : 3.8.1
GEOS lib   : /Users/christina.bogdan/miniconda3/envs/ovl/lib/libgeos_c.dylib
GDAL       : 3.1.4
GDAL data dir: /Users/christina.bogdan/miniconda3/envs/ovl/share/gdal
PROJ       : 7.2.0
PROJ data dir: /Users/christina.bogdan/miniconda3/envs/ovl/share/proj

PYTHON DEPENDENCIES
-------------------
geopandas  : 0.9.0
pandas     : 1.1.5
fiona      : 1.8.18
numpy      : 1.19.5
shapely    : 1.7.1
rtree      : 0.9.7
pyproj     : 3.0.0.post1
matplotlib : 3.3.4
mapclassify: None
geopy      : None
psycopg2   : None
geoalchemy2: None
pyarrow    : 3.0.0
pygeos     : 0.8

</details>

![image](https://user-images.githubusercontent.com/17129091/109728779-d981e400-7b84-11eb-9ce9-272844087d0b.png)

**Comment by jorisvandenbossche** [comment]
@christaina Thanks a lot for the report!

That's not good .. but I can also reproduce the issue. I suppose this might be caused by pyproj.

**Comment by jorisvandenbossche** [comment]
So it is indeed the pyproj `query_utm_crs_info` function that segfaults, specifically when feeding it an area of interest with NaNs (because the bounds of an empty GeoDataFrame given NaN bounds):

```
In [1]: import geopandas as gpd
   ...: 
   ...: gdf = gpd.GeoDataFrame([], columns={'geometry'}, crs='epsg:4326')

In [2]: gdf.total_bounds
Out[2]: array([nan, nan, nan, nan])

In [3]: minx, miny, maxx, maxy = gdf.total_bounds

In [4]: x_center = np.mean([minx, maxx])
   ...: y_center = np.mean([miny, maxy])

In [5]: from pyproj.aoi import AreaOfInterest

In [6]: area_of_interest = AreaOfInterest(
   ...:     west_lon_degree=x_center,
   ...:     south_lat_degree=y_center,
   ...:     east_lon_degree=x_center,
   ...:     north_lat_degree=y_center
   ...: )

In [7]: from pyproj.database import query_utm_crs_info

In [8]: query_utm_crs_info(datum_name="WGS 84", area_of_interest=area_of_interest)
Segmentation fault (core dumped)
```

@snowman2 we should maybe not call `query_utm_crs_info` for empty frames (but rather raise an informative error message), but I think we should also prevent pyproj itself from segfaulting

**Comment by jorisvandenbossche** [comment]
Opened https://github.com/pyproj4/pyproj/issues/788 on the pyproj side

**Comment by martinfleis** [comment]
Shall we mark this as an upstream issue? I don't think it is necessary to have a fix in geopandas if there'll be one in pyproj.

**Comment by jorisvandenbossche** [comment]
This is indeed an upstream issue in pyproj, and has been fixed there since pyproj 3.1. Thus closing here.
